### ISSUE TITLE
Unprompted causing "IndexError" when batch size>1 and hiresfix enabled

### ISSUE THREAD

**Issue by Dawgmastah** [ThereforeGames/unprompted#200]
### Due diligence

- [X] I checked for similar issues and couldn't find any.
- [X] My WebUI and Unprompted are both up-to-date.
- [X] I disabled my other extensions but the problem persists.

### Describe the bug

As stated, whenever unprompted is enabled and without even using any shortcodes, this error happens with batch  size >1 and hiresfix enabled.

### Prompt

irrelevant

### Log output

```shell
Traceback (most recent call last):
      File "X:\STABLEDIFFUSION\AUTOMATIC11111\modules\call_queue.py", line 57, in f
        res = list(func(*args, **kwargs))
      File "X:\STABLEDIFFUSION\AUTOMATIC11111\modules\call_queue.py", line 36, in f
        res = func(*args, **kwargs)
      File "X:\STABLEDIFFUSION\AUTOMATIC11111\modules\txt2img.py", line 55, in txt2img
        processed = processing.process_images(p)
      File "X:\STABLEDIFFUSION\AUTOMATIC11111\modules\processing.py", line 732, in process_images
        res = process_images_inner(p)
      File "X:\STABLEDIFFUSION\AUTOMATIC11111\extensions\sd-webui-controlnet\scripts\batch_hijack.py", line 42, in processing_process_images_hijack
        return getattr(processing, '__controlnet_original_process_images_inner')(p, *args, **kwargs)
      File "X:\STABLEDIFFUSION\AUTOMATIC11111\modules\processing.py", line 867, in process_images_inner
        samples_ddim = p.sample(conditioning=p.c, unconditional_conditioning=p.uc, seeds=p.seeds, subseeds=p.subseeds, subseed_strength=p.subseed_strength, prompts=p.prompts)
      File "X:\STABLEDIFFUSION\AUTOMATIC11111\modules\processing.py", line 1156, in sample
        return self.sample_hr_pass(samples, decoded_samples, seeds, subseeds, subseed_strength, prompts)
      File "X:\STABLEDIFFUSION\AUTOMATIC11111\modules\processing.py", line 1242, in sample_hr_pass
        samples = self.sampler.sample_img2img(self, samples, noise, self.hr_c, self.hr_uc, steps=self.hr_second_pass_steps or self.steps, image_conditioning=image_conditioning)
      File "X:\STABLEDIFFUSION\AUTOMATIC11111\modules\sd_samplers_kdiffusion.py", line 188, in sample_img2img
        samples = self.launch_sampling(t_enc + 1, lambda: self.func(self.model_wrap_cfg, xi, extra_args=self.sampler_extra_args, disable=False, callback=self.callback_state, **extra_params_kwargs))
      File "X:\STABLEDIFFUSION\AUTOMATIC11111\modules\sd_samplers_common.py", line 261, in launch_sampling
        return func()
      File "X:\STABLEDIFFUSION\AUTOMATIC11111\modules\sd_samplers_kdiffusion.py", line 188, in <lambda>
        samples = self.launch_sampling(t_enc + 1, lambda: self.func(self.model_wrap_cfg, xi, extra_args=self.sampler_extra_args, disable=False, callback=self.callback_state, **extra_params_kwargs))
      File "X:\STABLEDIFFUSION\AUTOMATIC11111\venv\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
        return func(*args, **kwargs)
      File "X:\STABLEDIFFUSION\AUTOMATIC11111\repositories\k-diffusion\k_diffusion\sampling.py", line 594, in sample_dpmpp_2m
        denoised = model(x, sigmas[i] * s_in, **extra_args)
      File "X:\STABLEDIFFUSION\AUTOMATIC11111\venv\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
        return forward_call(*args, **kwargs)
      File "X:\STABLEDIFFUSION\AUTOMATIC11111\modules\sd_samplers_cfg_denoiser.py", line 125, in forward
        x_in = torch.cat([torch.stack([x[i] for _ in range(n)]) for i, n in enumerate(repeats)] + [x])
      File "X:\STABLEDIFFUSION\AUTOMATIC11111\modules\sd_samplers_cfg_denoiser.py", line 125, in <listcomp>
        x_in = torch.cat([torch.stack([x[i] for _ in range(n)]) for i, n in enumerate(repeats)] + [x])
      File "X:\STABLEDIFFUSION\AUTOMATIC11111\modules\sd_samplers_cfg_denoiser.py", line 125, in <listcomp>
        x_in = torch.cat([torch.stack([x[i] for _ in range(n)]) for i, n in enumerate(repeats)] + [x])
    IndexError: index 1 is out of bounds for dimension 0 with size 1
```

### Unprompted version

v9.16.1

### WebUI version

1.6

**Comment by Dawgmastah** [comment]
Additionally, this is the output in unprompted 9.15

*** Error running process_batch: X:\STABLEDIFFUSION\AUTOMATIC11111\extensions\_unprompted\scripts\unprompted.py
    Traceback (most recent call last):
      File "X:\STABLEDIFFUSION\AUTOMATIC11111\extensions\_unprompted\scripts\unprompted.py", line 737, in process_batch
        if "extra_network_data" in p: extra_networks.deactivate(p, p.extra_network_data)
    TypeError: argument of type 'StableDiffusionProcessingTxt2Img' is not iterable

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "X:\STABLEDIFFUSION\AUTOMATIC11111\modules\scripts.py", line 643, in process_batch
        script.process_batch(p, *script_args, **kwargs)
      File "X:\STABLEDIFFUSION\AUTOMATIC11111\extensions\_unprompted\scripts\unprompted.py", line 739, in process_batch
        self.log.exception("Exception while deactiating extra networks")
    AttributeError: 'Scripts' object has no attribute 'log'


Error is different but images are generated

**Comment by ThereforeGames** [comment]
Hi @Dawgmastah,

Thank you for reaching out. In Unprompted v10.1.0, I am able to generate images with a `batch_size` of 2 and hires fix enabled.

Looking at your console output, it appears that the problem may be related to ControlNet. Have you tried without ControlNet enabled?

**Comment by dfjunk** [comment]
> Hi @Dawgmastah,
> 
> Thank you for reaching out. In Unprompted v10.1.0, I am able to generate images with a `batch_size` of 2 and hires fix enabled.
> 
> Looking at your console output, it appears that the problem may be related to ControlNet. Have you tried without ControlNet enabled?

Since they never responded, I figured I'd chip in here. I wonder if it's possible they meant "Batch count". I get "IndexError: index 1 is out of bounds for dimension 0 with size 1" whenever I try a batch count more than 1 when using hires.fix, and I don't use controlnet. I made sure adetailer was off too since I usually use that. It runs the first one fine, and then the second one runs fine right up until it switches to the hires.fix process. Hope this helps. Also sorry if the format below is wrong, I'm not very familiar with github.

> Traceback (most recent call last):
>       File "H:\stable-diffusion\stable-diffusion-webui\modules\call_queue.py", line 57, in f
>         res = list(func(*args, **kwargs))
>                    ^^^^^^^^^^^^^^^^^^^^^
>       File "H:\stable-diffusion\stable-diffusion-webui\modules\call_queue.py", line 36, in f
>         res = func(*args, **kwargs)
>               ^^^^^^^^^^^^^^^^^^^^^
>       File "H:\stable-diffusion\stable-diffusion-webui\modules\txt2img.py", line 55, in txt2img
>         processed = processing.process_images(p)
>                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>       File "H:\stable-diffusion\stable-diffusion-webui\modules\processing.py", line 732, in process_images
>         res = process_images_inner(p)
>               ^^^^^^^^^^^^^^^^^^^^^^^
>       File "H:\stable-diffusion\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\batch_hijack.py", line 42, in processing_process_images_hijack
>         return getattr(processing, '__controlnet_original_process_images_inner')(p, *args, **kwargs)
>                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>       File "H:\stable-diffusion\stable-diffusion-webui\modules\processing.py", line 867, in process_images_inner
>         samples_ddim = p.sample(conditioning=p.c, unconditional_conditioning=p.uc, seeds=p.seeds, subseeds=p.subseeds, subseed_strength=p.subseed_strength, prompts=p.prompts)
>                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>       File "H:\stable-diffusion\stable-diffusion-webui\modules\processing.py", line 1156, in sample
>         return self.sample_hr_pass(samples, decoded_samples, seeds, subseeds, subseed_strength, prompts)
>                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>       File "H:\stable-diffusion\stable-diffusion-webui\modules\processing.py", line 1242, in sample_hr_pass
>         samples = self.sampler.sample_img2img(self, samples, noise, self.hr_c, self.hr_uc, steps=self.hr_second_pass_steps or self.steps, image_conditioning=image_conditioning)
>                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>       File "H:\stable-diffusion\stable-diffusion-webui\modules\sd_samplers_kdiffusion.py", line 188, in sample_img2img
>         samples = self.launch_sampling(t_enc + 1, lambda: self.func(self.model_wrap_cfg, xi, extra_args=self.sampler_extra_args, disable=False, callback=self.callback_state, **extra_params_kwargs))
>                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>       File "H:\stable-diffusion\stable-diffusion-webui\modules\sd_samplers_common.py", line 261, in launch_sampling
>         return func()
>                ^^^^^^
>       File "H:\stable-diffusion\stable-diffusion-webui\modules\sd_samplers_kdiffusion.py", line 188, in <lambda>
>         samples = self.launch_sampling(t_enc + 1, lambda: self.func(self.model_wrap_cfg, xi, extra_args=self.sampler_extra_args, disable=False, callback=self.callback_state, **extra_params_kwargs))
>                                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>       File "H:\stable-diffusion\stable-diffusion-webui\venv\Lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
>         return func(*args, **kwargs)
>                ^^^^^^^^^^^^^^^^^^^^^
>       File "H:\stable-diffusion\stable-diffusion-webui\repositories\k-diffusion\k_diffusion\sampling.py", line 594, in sample_dpmpp_2m
>         denoised = model(x, sigmas[i] * s_in, **extra_args)
>                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>       File "H:\stable-diffusion\stable-diffusion-webui\venv\Lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
>         return forward_call(*args, **kwargs)
>                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>       File "H:\stable-diffusion\stable-diffusion-webui\modules\sd_samplers_cfg_denoiser.py", line 125, in forward
>         x_in = torch.cat([torch.stack([x[i] for _ in range(n)]) for i, n in enumerate(repeats)] + [x])
>                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>       File "H:\stable-diffusion\stable-diffusion-webui\modules\sd_samplers_cfg_denoiser.py", line 125, in <listcomp>
>         x_in = torch.cat([torch.stack([x[i] for _ in range(n)]) for i, n in enumerate(repeats)] + [x])
>                                       ^^^^^^^^^^^^^^^^^^^^^^^^
>       File "H:\stable-diffusion\stable-diffusion-webui\modules\sd_samplers_cfg_denoiser.py", line 125, in <listcomp>
>         x_in = torch.cat([torch.stack([x[i] for _ in range(n)]) for i, n in enumerate(repeats)] + [x])
>                                        ~^^^
>     IndexError: index 1 is out of bounds for dimension 0 with size 1

**Comment by unwize** [comment]
This issue occurs for me as well. 

i7 8700k, 
RTX 2080 (Gaming Driver)
32GB RAM

Up to on sdwebui and unprompted.

**Comment by ThereforeGames** [comment]
Thank you @dfjunk and @SlappedWithSilence for providing additional information.

I was able to confirm that the issue was indeed caused by hires fix and `batch_count` > 1, not `batch_size`. It seems hires fix logic may have changed in the latest WebUI update.

At any rate, I believe I have fixed it internally - a new version of Unprompted will be made available in the coming days.

**Comment by ThereforeGames** [comment]
Hi all,

This issue has been resolved in the newly released Unprompted v10.2.0.

**Comment by dfjunk** [comment]
> Hi all,
> 
> This issue has been resolved in the newly released Unprompted v10.2.0.

Thank you so much, it works great! I've been using this extension non-stop since I found it, and this fix makes it even better for my workflow. Just wanted to say a quick thank you again for all of this.

**Comment by ThereforeGames** [comment]
That's awesome to hear, @dfjunk. Thank you for the kind words!
